ios/MaasCore/Sources/MaasComponents/Components/BottomSheet/Alert/AlertBottomSheet.swift

```diff
@@ -0,0 +1,185 @@
+            image: AnyView? = nil,
+            spacing: Float = SpacingScale().xs,
+            foreground: Color? = nil,
+            background: Color? = nil,
+            action: @escaping () -> Void
+        ) {
+            self.text = text
+            self.image = image
+            self.spacing = spacing
+            self.foreground = foreground
+            self.background = background
+            self.action = action
+        }
+
+        public static func primary(title: String, image: AnyView? = nil, action: (() -> ())? = nil) -> Self {
+            .init(
+                title,
+                image: image,
+                action: { action?() }
+            )
+        }
+
+        public static func secondary(title: String, image: AnyView? = nil, action: (() -> ())? = nil) -> Self {
+            .init(
+                title,
+                image: image,
+                foreground: Color(.label),
+                background: Color(.systemGray6),
+                action: { action?() }
+            )
+        }
+
+        public static func cancel(title: String, image: AnyView? = nil, action: (() -> ())? = nil) -> Self {
+            .init(
+                title,
+                image: image,
+                foreground: Color(.label),
+                background: Color(.clear),
+                action: { action?() }
+            )
+        }
+    }
+}
+
+private extension AlertBottomSheet.Button {
+
+    func view(tapHandler: @escaping () -> ()) -> MaasComponents.Button {
+        .init(text, foreground: foreground, background: background, action: { tapHandler(); action() })
+    }
+}
+
+
+// MARK: - Presenting
+
+public extension View {
+
+    func alertBottomSheet(
+        error: Binding<ApiError?>,
+        image: UIImage? = nil,
+        title: String? = nil,
+        subtitle: String? = nil,
+        buttons: [AlertBottomSheet.Button] = [.cancel(title: "Cancel")]
+    ) -> some View {
+        // TODO: L10n | subtitle correct message
+        bottomSheet(
+            isPresented: .bool(error),
+            content: { AlertBottomSheet(isPresenting: .bool(error)) }
+        )
+        .environmentObject(
+            AlertBottomSheet.ViewModel(
+                image: image,
+                title: title ?? "Something went wrong",
+                subtitle: error.wrappedValue?.message ?? subtitle,
+                buttons: buttons
+            )
+        )
+    }
+
+    func alertBottomSheet(
+        isPresented: Binding<Bool>,
+        image: UIImage? = nil,
+        title: String? = nil,
+        subtitle: String? = nil,
+        buttons: [AlertBottomSheet.Button]
+    ) -> some View {
+
+        bottomSheet(
+            isPresented: isPresented,
+            content: { AlertBottomSheet(isPresenting: isPresented) }
+        )
+        .environmentObject(
+            AlertBottomSheet.ViewModel(image: image, title: title, subtitle: subtitle, buttons: buttons)
+        )
+    }
+}
+// TODO: Is this good ?
```

Comment:
I'd say yes 🙂 I'd just name it `developerMessage` and expose it from within the `MaasCore` package.